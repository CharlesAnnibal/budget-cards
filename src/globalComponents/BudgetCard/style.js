
import { StyleSheet } from 'react-native';


export let styles = StyleSheet.create({
  card: {
    width: '47%',
    marginHorizontal: '1.5%',
    marginVertical: '3%',
    height: 70,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 10,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    elevation:2,

  },
  flexbox:{
    display:'flex',
    height:60,
    width:'100%',
    flexDirection:'column',
    justifyContent: 'flex-start',
    alignItems:'flex-start',
    textAlign:'left'
  },
  colorIdentifierRow: {
    flexDirection:'row',
    flex:1
  },
  textRow:{
    display:'flex',
    flexDirection:'row', 
    flex:1,
  },  
  progressBarRow: {
    flex:1,
    justifyContent:'center',
    width:'100%',
    paddingHorizontal:'10%'
  },
 
  text: {
    flex:1,
    textAlign:'center',
    fontSize: 16,
    color: '#001489',
    paddingHorizontal:'5%'
    /*textShadowColor: '#000',
    textShadowOffset: {width: -1, height: 1},
    textShadowRadius: 3*/
  }
})


export function identifier(color) {
  let styles = {
    backgroundColor: color,
    width: '20%',
    height:'80%',
    borderColor:'yellow',
    borderTopLeftRadius: 15,
    borderTopRightRadius: 3,
    borderBottomLeftRadius: 3,
    borderBottomRightRadius: 15
  }
  return styles;
}