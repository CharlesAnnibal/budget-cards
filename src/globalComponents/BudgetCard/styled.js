import styled from 'styled-components/native';

export const Wrapper = styled.View`
      display:flex;
      width:100%; 
      justify-content:flex-end;
`

export const ProgressBar = styled.View`
    width:90%;
    margin:0 5%;
`