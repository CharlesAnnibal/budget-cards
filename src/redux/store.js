import { createStore, applyMiddleware } from "redux";
import { combineReducers } from "redux";
import budgets from "../redux/ducks/budgets";
import contextMenu from "../redux/ducks/contextMenu"
import {helloSaga} from "../sagas/main"
import {getAllBudgets} from "../sagas/budgetList"
import createSagaMiddleware from 'redux-saga'


console.log("antes de juntar os reducers:", {contextMenu:contextMenu, budgets:budgets})

const rootReducer = combineReducers({contextMenu, budgets });

const sagaMiddleware = createSagaMiddleware()

const store = createStore(
    rootReducer,
    applyMiddleware(sagaMiddleware)
);


const sagas = [
    helloSaga,
    getAllBudgets
]

sagas.forEach((saga)=>{
    sagaMiddleware.run(saga)
  })

export default store;