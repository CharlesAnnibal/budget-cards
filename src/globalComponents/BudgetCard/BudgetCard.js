import React, { Component } from 'react';
import { View, Text, TouchableHighlight } from 'react-native'
import * as Progress from 'react-native-progress';
import { styles, identifier } from './style';

export default class BudgetCard extends Component {
    constructor(props) {
        super(props);

        this.state = {
            amount: 50,
            valueUtilized: 0.3,
            colorIdentifier: identifier(props.color)
        }

        this.pressButton = this.pressButton.bind(this);
    }

    pressButton(event) {
        console.log(event.nativeEvent.pageX)
        this.props.showMenu(
            {
                id: this.props.id,
                positionX: event.nativeEvent.pageX,
                positionY: event.nativeEvent.pageY
            }
        )
    }

    render() {
        return (
            <TouchableHighlight
                onPress={(event) => this.pressButton(event)}
                underlayColor="#e6f1f7"
                style={styles.card}
            >

                <View style={styles.flexbox}>
                    <View style={styles.colorIdentifierRow}>
                        <View style={this.state.colorIdentifier}></View>
                    </View>
                    <View style={styles.textRow}>
                        <Text style={styles.text}>{this.props.description}</Text>
                        <Text style={styles.text}>{this.props.amount}</Text>
                    </View>
                    <View style={styles.progressBarRow}>
                        <Progress.Bar style={{ marginTop: '4.5%', width: '100%' }} progress={this.props.valueUtilized} height={10} />
                    </View>
                </View>
            </TouchableHighlight>
        );
    }
}