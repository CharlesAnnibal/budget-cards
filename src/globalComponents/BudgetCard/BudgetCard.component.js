import React, { useState } from 'react';
import { View, Text, TouchableHighlight } from 'react-native'
import * as Progress from 'react-native-progress';
import { styles, identifier } from './BudgetCard.style';
import { useDispatch, useSelector } from 'react-redux';
import { showMenu, hideMenu } from "../../redux/actions";

const BudgetCard = (props) => {
  const [id, setId] = useState(props.id);
  const [amount, setAmout] = useState(50);
  const [description, setDescription] = useState(props.description);
  const [valueUtilized, setValueUtilized] = useState(0.3);
  //const [styles, setStyles] = useState(styles);
  const [colorIdentifier, setColorIdentifier] = useState(identifier(props.color));


  const dispatch = useDispatch();
  const currentstate = useSelector(state => state.contextMenu);
  console.log("ue", currentstate)
  const budgets = useSelector(state => state.budgets);
    console.log("using selector budgets:::",budgets)

  const _onPressButton = () => {
    
    dispatch(showMenu(id));
    //useDispatch(hideMenu(id));

    //this.props.navigate.navigate('NewExpense')
    //this.menuRef.current.toggleMenu()
    //this.state.showingMenu ? this.setState({ showingMenu: false }) : this.setState({ showingMenu: true })
  }

  return (
    <TouchableHighlight onPress={() => _onPressButton()} style={styles.card}>
      <View style={styles.flexbox}>
        <View style={styles.colorIdentifierRow}>
          <View style={colorIdentifier}></View>
        </View>
        <View style={styles.textRow}>
          <Text style={styles.text}>{description}</Text>
          <Text style={styles.text}>{amount}</Text>
        </View>
        <View style={styles.progressBarRow}>
          <Progress.Bar style={{ marginTop: '4.5%', width: '100%' }} progress={valueUtilized} height={10} />
        </View>
      </View>
    </TouchableHighlight>
  );

}

export default BudgetCard;

