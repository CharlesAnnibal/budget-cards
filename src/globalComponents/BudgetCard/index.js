import { connect } from 'react-redux';
import { showMenu } from "../../redux/ducks/contextMenu";

import BudgetCard from './BudgetCard'
export default connect(null, {showMenu})(BudgetCard);




