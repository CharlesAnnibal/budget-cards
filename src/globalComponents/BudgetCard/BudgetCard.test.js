import React from 'react'
import { mount, shallow } from 'enzyme'
import BudgetCard from '.'
import store from "../../redux/store";


describe('<BudgetCard />', () => {
    const props = {
        budgets: [],
        id: 1,
        description: "Agua",
        color: "red",
        store: store,
        //showMenu: jest.fn()
    }

    const wrapper = shallow(<BudgetCard {...props} />)

    it('renders correctly, test using Jest + Enzyme', () => {
        expect(wrapper).toMatchSnapshot();
    });

})