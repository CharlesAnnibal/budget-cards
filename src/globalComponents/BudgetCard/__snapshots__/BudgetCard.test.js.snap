// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BudgetCard /> renders correctly, test using Jest + Enzyme 1`] = `ShallowWrapper {}`;
